Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsSubComponentModeLinearFloat.cpp

```diff
@@ -0,0 +1,133 @@
+        {
+            float m_startingValue = 0.0f;
+        };
+        auto sharedState = AZStd::make_shared<SharedState>();
+
+        m_manipulator->InstallLeftMouseDownCallback(
+            [this, sharedState, idPair]([[maybe_unused]]const AzToolsFramework::LinearManipulator::Action& action) mutable
+            {
+                float currentValue = 0.0f;
+
+                EditorJointRequestBus::EventResult(currentValue, idPair, &EditorJointRequests::GetLinearValue, m_propertyName);
+                sharedState->m_startingValue = currentValue;
+            });
+
+        m_manipulator->InstallMouseMoveCallback(
+            [this, sharedState, idPair](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                const float axisDisplacement = action.LocalPositionOffset().Dot(action.m_fixed.m_axis);
+
+                float newValue = AZ::GetClamp(sharedState->m_startingValue + DisplacementToDeltaValue(axisDisplacement), m_min, m_max);
+                EditorJointRequestBus::Event(idPair, &EditorJointRequests::SetLinearValue, m_propertyName, newValue);
+
+                const AZ::Vector3 localPosition = action.LocalPosition().GetMax(AZ::Vector3(0.01f, 0.0f, 0.0f));
+                m_manipulator->SetLocalTransform(AZ::Transform::CreateTranslation(localPosition));
+            });
+
+        m_manipulator->Register(AzToolsFramework::g_mainManipulatorManagerId);
+    }
+
+    void JointsSubComponentModeLinearFloat::Refresh(const AZ::EntityComponentIdPair& idPair)
+    {
+        float currentValue = 0.0f;
+        EditorJointRequestBus::EventResult(currentValue, idPair, &EditorJointRequests::GetLinearValue, m_propertyName);
+
+        m_manipulator->SetLocalTransform(AZ::Transform::CreateTranslation(AZ::Vector3::CreateAxisX() * ValueToDisplacement(currentValue)));
+    }
+
+    void JointsSubComponentModeLinearFloat::Teardown(const AZ::EntityComponentIdPair& idPair)
+    {
+        m_manipulator->RemoveEntityComponentIdPair(idPair);
+        m_manipulator->Unregister();
+        m_manipulator.reset();
+    }
+
+    void JointsSubComponentModeLinearFloat::ResetValues(const AZ::EntityComponentIdPair& idPair)
+    {
+        EditorJointRequestBus::Event(idPair, &EditorJointRequests::SetLinearValue, m_propertyName, m_resetValue);
+        m_manipulator->SetLocalTransform(AZ::Transform::CreateTranslation(AZ::Vector3::CreateAxisX() * ValueToDisplacement(m_resetValue)));
+    }
+
+    float JointsSubComponentModeLinearFloat::DisplacementToDeltaValue(float displacement) const
+    {
+        if (displacement > 0.0f)
+        {
+            return powf(displacement, m_exponent);
+        }
+        else if (displacement < 0.0f)
+        {
+            return -powf(fabsf(displacement), m_exponent);
+        }
+        else
+        {
+            return 0.0f;
+        }
+    }
+
+    float JointsSubComponentModeLinearFloat::ValueToDisplacement(float value) const
+    {
+        return powf(value, m_inverseExponent);
```

Comment:
nit: You can use `AZStd::pow` not (that's the C++ version and is overloaded for floats do you don't need the trailing `f` 🙂